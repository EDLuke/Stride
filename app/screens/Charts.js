import React from 'react';
import { FlatList, ScrollView, StyleSheet, Text, View, ToolbarAndroid } from 'react-native';
import { Container, Button, Content, List } from 'native-base';
import User from '../components/class/UserClass.js';
import { SmoothLine, StockLine } from 'react-native-pathjs-charts';
import SingleCalorie from '../components/layout/SingleCalorie';
import moment from 'moment';
import { Octicons, MaterialIcons } from '@expo/vector-icons';
import { Toolbar } from 'react-native-material-ui';

GLOBAL = require('../components/CurrentUser');

const renderCalorie = (post, index) => (
	<SingleCalorie
	    key={index}
	    date={post.date}
	    calorie={post.calorie}
  	/>
)

export default class Charts extends React.Component{

	refresh = () => {
		// refresh the state
		this.setState({
			records: GLOBAL.currentUser.FitnessRecord.map(
			    	function(fitness){
			    		var first = GLOBAL.currentUser.FitnessRecord[0];

			    		return {
			    			days: moment(fitness.date).diff(first.date, 'days'),
			    			calorie: parseInt(fitness.calorie),
			    		}
	    			}),
		});
	}

	state = {
	    records: GLOBAL.currentUser.FitnessRecord.map(
	    	function(fitness){
	    		var first = GLOBAL.currentUser.FitnessRecord[0];

	    		return {
	    			days: moment(fitness.date).diff(first.date, 'days'),
	    			calorie: parseInt(fitness.calorie),
	    		}
	    	}),
	    options: {
	      width: 250,
	      height: 250,
	      color: '#2980B9',
	      margin: {
	        top: 0,
	        left: 45,
	        bottom: 35,
	        right: 25,
	      },
	      animate: {
	        type: 'delayed',
	        duration: 200
	      },
	      axisX: {
	        showAxis: true,
	        showLines: true,
	        showLabels: true,
	        showTicks: true,
	        zeroAxis: false,
	        orient: 'bottom',
	        tickValues: [],
	        labelFunction: ((v) => {
	          let d = moment(GLOBAL.currentUser.FitnessRecord[0].date)

	          if(v % 1 === 0)
	          	return d.add(v,'days').format('MM-DD') 
	          return ""
	        }),
	        label: {
	          fontFamily: 'Arial',
	          fontSize: 8,
	          fontWeight: true,
	          fill: '#34495E'
	        }
	      },
	      axisY: {
	        showAxis: true,
	        showLines: true,
	        showLabels: true,
	        showTicks: true,
	        zeroAxis: false,
	        orient: 'left',
	        tickValues: [],
	        label: {
	          fontFamily: 'Arial',
	          fontSize: 8,
	          fontWeight: true,
	          fill: '#34495E'
	        }
	      }
	  	}
	};

	onPressAdd = () => {
		// Navigate to the ChartsAdd
        this.props.navigation.navigate('ChartsAdd', {refresh: this.refresh});
	}

	render() {
	    let data = [
	      this.state.records
	    ]


	    let options = this.state.options
    
		return (
			<View style={styles.container}>
				<Toolbar
					centerElement="Progression"
			    />

			    <View style={styles.chartContainer}>
	       		 <StockLine data={data} options={options} xKey='days' yKey='calorie' />
			    </View>
			    <ScrollView style={styles.listContainer}>			
	          		<List>
		            {
		              this.state.records.map((record, index) => {
		              	let d = moment(GLOBAL.currentUser.FitnessRecord[0].date)

			          	let date = d.add(record.days,'days').format('MM-DD')
			          	let recordNew = {
			          		date: date,
			          		calorie: record.calorie.toString()
			          	}

		              	return renderCalorie(recordNew, index)
		              })
		            }
		          	</List>
			    </ScrollView>
		      	
		      	<Button
		          rounded
		          style={styles.button}
		          onPress={() => this.onPressAdd()}
		        >
		        	<Octicons
		            name="plus"
		            size={16}
		            color="#FFF"
		            style=
		            {{
		            	left: 2,
		            	padding: 15
		            }}
		          />
		        </Button>
	      </View>
		);
	}
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  chartContainer: {
  	marginTop: 10,
  	alignItems: 'center',
    justifyContent: 'center',
  },
  listContainer: {

  },
  toolbar: {
    backgroundColor: '#578CA9',
    height: 56,
  },
  button: {
    position: 'absolute',
    bottom: 20,
    right: 20,
  },
});
