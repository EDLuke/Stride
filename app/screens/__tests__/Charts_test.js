import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import Charts from '../Charts';
import { View } from 'react-native';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const wrapper = shallow(
      <Charts/>
    );
  expect(wrapper).toMatchSnapshot();
});