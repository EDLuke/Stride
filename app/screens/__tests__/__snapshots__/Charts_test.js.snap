// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <SmoothLineChart
    data={
      Array [
        Array [
          Object {
            "x": -10,
            "y": -1000,
          },
          Object {
            "x": -9,
            "y": -729,
          },
          Object {
            "x": -8,
            "y": -512,
          },
          Object {
            "x": -7,
            "y": -343,
          },
          Object {
            "x": -6,
            "y": -216,
          },
          Object {
            "x": -5,
            "y": -125,
          },
          Object {
            "x": -4,
            "y": -64,
          },
          Object {
            "x": -3,
            "y": -27,
          },
          Object {
            "x": -2,
            "y": -8,
          },
          Object {
            "x": -1,
            "y": -1,
          },
          Object {
            "x": 0,
            "y": 0,
          },
          Object {
            "x": 1,
            "y": 1,
          },
          Object {
            "x": 2,
            "y": 8,
          },
          Object {
            "x": 3,
            "y": 27,
          },
          Object {
            "x": 4,
            "y": 64,
          },
          Object {
            "x": 5,
            "y": 125,
          },
          Object {
            "x": 6,
            "y": 216,
          },
          Object {
            "x": 7,
            "y": 343,
          },
          Object {
            "x": 8,
            "y": 512,
          },
          Object {
            "x": 9,
            "y": 729,
          },
          Object {
            "x": 10,
            "y": 1000,
          },
        ],
        Array [
          Object {
            "x": -10,
            "y": 100,
          },
          Object {
            "x": -9,
            "y": 81,
          },
          Object {
            "x": -8,
            "y": 64,
          },
          Object {
            "x": -7,
            "y": 49,
          },
          Object {
            "x": -6,
            "y": 36,
          },
          Object {
            "x": -5,
            "y": 25,
          },
          Object {
            "x": -4,
            "y": 16,
          },
          Object {
            "x": -3,
            "y": 9,
          },
          Object {
            "x": -2,
            "y": 4,
          },
          Object {
            "x": -1,
            "y": 1,
          },
          Object {
            "x": 0,
            "y": 0,
          },
          Object {
            "x": 1,
            "y": 1,
          },
          Object {
            "x": 2,
            "y": 4,
          },
          Object {
            "x": 3,
            "y": 9,
          },
          Object {
            "x": 4,
            "y": 16,
          },
          Object {
            "x": 5,
            "y": 25,
          },
          Object {
            "x": 6,
            "y": 36,
          },
          Object {
            "x": 7,
            "y": 49,
          },
          Object {
            "x": 8,
            "y": 64,
          },
          Object {
            "x": 9,
            "y": 81,
          },
          Object {
            "x": 10,
            "y": 100,
          },
        ],
      ]
    }
    options={
      Object {
        "animate": Object {
          "duration": 200,
          "type": "delayed",
        },
        "axisX": Object {
          "label": Object {
            "fill": "#34495E",
            "fontFamily": "Arial",
            "fontSize": 14,
            "fontWeight": true,
          },
          "orient": "bottom",
          "showAxis": true,
          "showLabels": true,
          "showLines": true,
          "showTicks": true,
          "zeroAxis": false,
        },
        "axisY": Object {
          "label": Object {
            "fill": "#34495E",
            "fontFamily": "Arial",
            "fontSize": 14,
            "fontWeight": true,
          },
          "orient": "left",
          "showAxis": true,
          "showLabels": true,
          "showLines": true,
          "showTicks": true,
          "zeroAxis": false,
        },
        "color": "#2980B9",
        "height": 280,
        "margin": Object {
          "bottom": 25,
          "left": 45,
          "right": 20,
          "top": 20,
        },
        "width": 280,
      }
    }
    xKey="x"
    yKey="y"
  />
</View>
`;
